File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if len(nums) < 2:
            return nums
        
        mid = len(nums) // 2
        left = nums[:mid]
        right = nums[mid:]

        leftSorted = self.sortArray(left)
        rightSorted = self.sortArray(right)

        l = 0
        r = 0
        res = []

        while (l < len(leftSorted)) and (r < len(rightSorted)):
            if (leftSorted[l] <= rightSorted[r]):
                res.append(leftSorted[l])
                l += 1
            else:
                res.append(rightSorted[r])
                r += 1

        if l < len(leftSorted):
            ptr = l
            remaining = leftSorted
        else:
            ptr = r
            remaining = rightSorted

        for i in range(ptr, len(remaining)):
            res.append(remaining[i])

        return res

File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = nums.size();
        int i = 0;

        while (i < nums.size()) {
            if (nums[i] == val) {
                k--;
                nums.erase(nums.begin() + i);
            } else {
                i++;
            }
        }

        return k;
    }
};